### DISCUSSION TITLE
Versions drafts autosave enabled .. slug not generating

### DISCUSSION THREAD

**Discussion by pcbal** [payloadcms/payload#2175] [Q&A]
Slug for the page or post not created automatically on save when auto save drafts versions enabled. autosave off enables save draft button and on clicking it slug field filled up as normal. 
![05:44:32_24_02_2023_*](https://user-images.githubusercontent.com/40548855/221061297-cde2b3c5-f929-4c9d-858e-6a746552e04b.jpg)

**Comment by jmikrut** [comment]
Hey @pcbal — I believe this is likely because you are using a slug generator that you found in Discord or one of our own projects, maybe something like this:

```
const format = (val) => val.replace(/ /g, '-').replace(/[^\w-]+/g, '').toLowerCase();

const slugBeforeValidateHook = ({ operation, value, originalDoc, data }) => {
  if (typeof value === 'string') {
    return format(value);
  }

  if (operation === 'create') {
    const fallbackData = (data && data.title) || (originalDoc && originalDoc.title);

    if (fallbackData && typeof fallbackData === 'string') {
      return format(fallbackData);
    }
  }

  return value;
}
```

Thing is that this hook _**only**_ works on `create` operations, and is intended to be used on `draft` collections but NOT autosave collections.

Autosave collections "create" instantly as soon as you click `Create new` - so at that point, you will have no title field / whatever field you are using to generate your slug. If you want to use something like this for autosave as well, you'll need to modify the logic. Maybe make it so that instead of  the hook says `if (operation === 'create')`, it says `if (typeof value === 'undefined')`.

I haven't tested that but this is not a problem with Payload and is instead just a problem with the slug hook you've got. So I'll convert this to a discussion and we can keep the convo going there!

**Reply by pcbal** [reply]
Suggestion working when hit publish. But preview still affected without having a slug.
``// if (operation === 'create') {
	  if (typeof value === 'undefined') {
    const fallbackData = (data && data[fallback]) || (originalDoc && originalDoc[fallback]);

![10:17:02_27_02_2023_*](https://user-images.githubusercontent.com/40548855/221476718-35693462-bc7e-4a5a-b880-bdfc56e994f4.jpg)
